Library/Homebrew/cask/lib/hbc/cask.rb

```diff
@@ -85,6 +85,20 @@ def installed?
       !versions.empty?
     end
 
+    def latest_installed_version
+      versions.last
+    end
+
+    def has_source?()
+      !version.nil?
+    end
+
+    def outdated?(greedy = false)
+      has_source? &&
+      (greedy || !auto_updates && !version == :latest) &&
+      version != latest_installed_version.to_s
```

Comment:
You can remove `to_s` here after https://github.com/Homebrew/brew/pull/2282 is merged.